hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -538,6 +544,23 @@ public Pair<AbfsRestOperation, Boolean> renamePath(
         if (!op.hasResult()) {
           throw e;
         }
+
```

Comment:
have renamePath return a struct rather than a pair, with one of the fields being the "retried for metadata issue" flag. this can be passed up to the private resilient rename interface, and then to the manifest committer. it could then include this in its statistic reports